packages/vscode-mdx/src/typescript-plugin.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * @typedef {import('typescript/lib/tsserverlibrary.js').server.Project} Project
+ * @typedef {import('typescript/lib/tsserverlibrary.js').server.PluginModuleFactory} PluginModuleFactory
+ * @typedef {import('typescript/lib/tsserverlibrary.js').LanguageService} LanguageService
+ */
+
+import {
+  decorateLanguageService,
+  decorateLanguageServiceHost,
+  searchExternalFiles
+} from '@volar/typescript'
+import {createFileProvider, resolveCommonLanguageId} from '@volar/language-core'
+import {getLanguageModule} from '@mdx-js/language-service'
+import {loadPlugins} from '../../language-server/lib/configuration.js'
+
+/**
+ * @type {WeakMap<Project, string[]>}
+ */
+const externalFiles = new WeakMap()
+
+/**
+ * @type {PluginModuleFactory} modules
+ */
+const init = (modules) => {
+  const {typescript: ts} = modules
+  return {
+    create(info) {
+      const enptySnapshot = ts.ScriptSnapshot.fromString('')
+      const getScriptSnapshot = info.languageServiceHost.getScriptSnapshot.bind(info.languageServiceHost)
+      const getScriptVersion = info.languageServiceHost.getScriptVersion.bind(info.languageServiceHost)
+      const getScriptKind = info.languageServiceHost.getScriptKind?.bind(info.languageServiceHost)
+      const getProjectVersion = info.languageServiceHost.getProjectVersion?.bind(info.languageServiceHost)
+      const configFileName = info.project.projectKind === ts.server.ProjectKind.Configured
+        ? info.project.getProjectName()
+        : undefined
+
+      info.languageServiceHost.getScriptSnapshot = (fileName) => {
+        if (!initialized && fileName.endsWith('.mdx')) {
+          return enptySnapshot
+        }
+        return getScriptSnapshot(fileName)
+      }
+      info.languageServiceHost.getScriptVersion = (fileName) => {
+        if (!initialized && fileName.endsWith('.mdx')) {
+          return 'initializing...'
```

Comment:
I think this can be any string, as long as it’s not numeric, right?

```suggestion
          return 'initializing…'
```

Reply:
Yes.